src/User/AutoLoginMiddleware.php

```diff
@@ -1,16 +1,91 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Yiisoft\Yii\Web\User;
 
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Log\LoggerInterface;
+use Yiisoft\Auth\IdentityRepositoryInterface;
+
 /**
  * AutoLoginMiddleware automatically logs user in based on "remember me" cookie
  */
-class AutoLoginMiddleware
+final class AutoLoginMiddleware implements MiddlewareInterface
 {
     private User $user;
+    private IdentityRepositoryInterface $identityRepository;
+    private LoggerInterface $logger;
 
-    public function __construct(User $user)
-    {
+    public function __construct(
+        User $user,
+        IdentityRepositoryInterface $identityRepository,
+        LoggerInterface $logger
+    ) {
         $this->user = $user;
+        $this->identityRepository = $identityRepository;
+        $this->logger = $logger;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $response = $handler->handle($request);
```

Comment:
Shouldn't we try to log user in _before_ handling request?

Reply:
It is true! I have a question, how can I get the response? For example DI in the method process?

Reply:
You don't need response in this case but, in general, you create it via response factory. See https://github.com/yiisoft/yii-web/blob/master/src/Middleware/Csrf.php#L40

Reply:
Aha, I think I've got what you've meant...

Reply:
The method body should be something like:

```php
if (!$this->authenticateUserFromRequest($request, $response)) {
    $this->logger->warning('Unable to authenticate user by cookie.');
}

return $handler->handle($request);
```

Reply:
But I need add the response for save the cookie. I will add the response with the responseFactory, but i have a question. This new response it is different to the response of the handler, right?. Is it a problem?  

```
$response = $this->responseFactory->createResponse(Status::UNPROCESSABLE_ENTITY);
if (!$this->authenticateUserFromRequest($request, $response)) {
    $this->logger->warning('Unable to authenticate user by cookie.');
}

return $handler->handle($request);
```

Reply:
> This new response it is different to the response of the handler, right?. Is it a problem?

Yes, that's a problem.

Reply:
It seems like sending identity cookie should be done separately from logging in by cookie.

Reply:
How would this be?